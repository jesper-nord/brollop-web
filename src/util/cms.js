export const GRAPHCMS_ENDPOINT = 'https://api-eu-central-1.graphcms.com/v2/ckc7ycwra09i001xr559jgvds/master';

export const getPageContentQuery = id => `
    {
      pageContent(where: { 
        id: "${id}" 
      }) {
        textContent {
          html
        }
      }
    }
  `;
