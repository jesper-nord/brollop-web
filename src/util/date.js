import {
  isAfter, differenceInMilliseconds, parseISO, format
} from 'date-fns';

const DAY_MS = 24 * 60 * 60 * 1000;
const HOUR_MS = 60 * 60 * 1000;

const millisecondsToDays = ms => {
  let days = Math.floor(ms / DAY_MS);
  let hours = Math.floor((ms - days * DAY_MS) / HOUR_MS);
  let minutes = Math.round((ms - days * DAY_MS - hours * HOUR_MS) / 60000);
  if (minutes === 60) {
    hours += 1;
    minutes = 0;
  }
  if (hours === 24) {
    days += 1;
    hours = 0;
  }
  return { days, hours, minutes };
};

export {
  isAfter,
  differenceInMilliseconds,
  parseISO,
  format,
  millisecondsToDays
};
