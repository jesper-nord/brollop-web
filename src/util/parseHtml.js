import ReactHtmlParser from 'react-html-parser';

export const parseHtml = html => ReactHtmlParser(html);
