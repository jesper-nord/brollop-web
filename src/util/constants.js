import { parseISO } from './date';

export const GA_KEY = 'UA-159992142-1';
export const WEDDING_DATE = parseISO('2020-09-05T14:30:00');
