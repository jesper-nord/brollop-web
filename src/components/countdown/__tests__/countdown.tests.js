import React from 'react';
import { render } from '@testing-library/react';
import MockDate from 'mockdate';
import { parseISO } from '../../../util/date';
import { Countdown } from '..';

describe('countdown', () => {
  test('renders countdown', async () => {
    MockDate.set(parseISO('2025-03-01T09:00:00'));
    const toDate = parseISO('2025-03-25T14:30:00');

    const { container, queryByText } = render(<Countdown to={toDate} />);

    expect(container).toMatchSnapshot();
    expect(queryByText('24')).toBeInTheDocument();
    expect(queryByText('dagar')).toBeInTheDocument();
    expect(queryByText('5')).toBeInTheDocument();
    expect(queryByText('timmar')).toBeInTheDocument();
    expect(queryByText('30')).toBeInTheDocument();
    expect(queryByText('minuter')).toBeInTheDocument();
  });

  test('renders without days', async () => {
    MockDate.set(parseISO('2025-03-25T09:00:00'));
    const toDate = parseISO('2025-03-25T14:30:00');

    const { container, queryByText } = render(<Countdown to={toDate} />);

    expect(container).toMatchSnapshot();
    expect(queryByText('dagar')).toBe(null);
    expect(queryByText('5')).toBeInTheDocument();
    expect(queryByText('timmar')).toBeInTheDocument();
    expect(queryByText('30')).toBeInTheDocument();
    expect(queryByText('minuter')).toBeInTheDocument();
  });

  test('renders without hours', async () => {
    MockDate.set(parseISO('2025-03-25T14:00:00'));
    const toDate = parseISO('2025-03-25T14:30:00');

    const { container, queryByText } = render(<Countdown to={toDate} />);

    expect(container).toMatchSnapshot();
    expect(queryByText('dagar')).toBe(null);
    expect(queryByText('timmar')).toBe(null);
    expect(queryByText('30')).toBeInTheDocument();
    expect(queryByText('minuter')).toBeInTheDocument();
  });
});
