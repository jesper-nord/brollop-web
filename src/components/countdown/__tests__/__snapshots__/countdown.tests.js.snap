// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countdown renders countdown 1`] = `
<div>
  <div
    class="countdown"
  >
    <div
      class="unit"
    >
      <p
        class="header"
      >
        24
      </p>
      <p>
        dagar
      </p>
    </div>
    <div
      class="unit"
    >
      <p
        class="header"
      >
        5
      </p>
      <p>
        timmar
      </p>
    </div>
    <div
      class="unit"
    >
      <p
        class="header"
      >
        30
      </p>
      <p>
        minuter
      </p>
    </div>
  </div>
</div>
`;

exports[`countdown renders without days 1`] = `
<div>
  <div
    class="countdown"
  >
    <div
      class="unit"
    >
      <p
        class="header"
      >
        5
      </p>
      <p>
        timmar
      </p>
    </div>
    <div
      class="unit"
    >
      <p
        class="header"
      >
        30
      </p>
      <p>
        minuter
      </p>
    </div>
  </div>
</div>
`;

exports[`countdown renders without hours 1`] = `
<div>
  <div
    class="countdown"
  >
    <div
      class="unit"
    >
      <p
        class="header"
      >
        30
      </p>
      <p>
        minuter
      </p>
    </div>
  </div>
</div>
`;
